grouper/fe/util.py

```diff
@@ -264,3 +268,42 @@ def _decorator(self, *args, **kwargs):
         return wraps(f)(_decorator)
 
     return _wrapper
+
+
+def paginate_results(handler, results):
+    # type: (GrouperHandler, List[T]) -> tuple[int, int, int, List[T]]
+    """Limits the number of results to display for handlers/templates the paginate lists
+
+    Args:
+        handler: the GrouperHandler for the request being paginated
+        results: the entire list of possible results
+
+    Returns:
+        the total number of results, the offset, the limit, and the limited results to show
+            for this page
+    """
+    total = len(results)
+    offset = int(handler.get_argument("offset", 0))
+    limit = int(handler.get_argument("limit", 100))
+    if limit > 9000:
+        limit = 9000
+
+    return total, offset, limit, results[offset:offset + limit]
+
+
+def form_http_verbs(function):
+    """This decorator is used to support forms submitted using POST as if they were
+    submitted using less common HTTP verbs, such as delete. After annotating the
+    post function in the handler, all forms that have the appropriate field
+    (HTTP_VERB) will have their requests routed through the appropriate function
+    instead of the annotated function. This allows us to use these HTTP verbs
+    where possible, and then fall back to using forms via POST while still using
+    the same code path. See the http_verb macro in macros/ui.html for implementing
+    the form field.
+    """
+    def inner(self, *args, **kwargs):
+        if "HTTP_VERB" in self.request.arguments:
+            func = getattr(self, self.request.arguments["HTTP_VERB"][0].lower())
```

Comment:
We should whitelist the verbs we'll accept, otherwise I could submit something like `_my_internal_function` and maybe do something unexpected.